### ISSUE TITLE
AtrousConvolution2D inconsistent output shape

### ISSUE THREAD

**Issue by DonghyunK** [fchollet/keras#5024]
I tried to use AtrousConvolution2D, but its behavior is not consistent.

---
from keras.models import Sequential
from keras.layers import ZeroPadding2D, AtrousConvolution2D
from keras import callbacks
import numpy as np

model = Sequential()
model.add(ZeroPadding2D((1,1), input_shape=(3, 12, 12), dim_ordering='th'))
model.add(AtrousConvolution2D(3, 3, 3,  atrous_rate=(2,2)))
model.summary()
model.compile(loss='mse', optimizer='adam')

print 'Actual Output Shape: ', model.predict(np.ones((1, 3, 12, 12))).shape
# Actual Output Shape:  (1, 3, 10, 10) <- as model.summary() says.

print 'Training completed successfully?:', isinstance(model.fit(np.ones((1,3,12,12)), np.ones((1,3,10,10)), nb_epoch=1, verbose=0), callbacks.History)
# Training completed successfully?: True

---
![image](https://cloud.githubusercontent.com/assets/20954382/21917633/19d2e2e4-d8fe-11e6-8306-78a211fe84ff.png)

As model.summary says that output of AtrousConvolution2D should be (1, 3, 10, 10), but actual shape is  (1, 3, 12, 12) 

Therefore, I cannot train the model with AtrousConvolution2D.

Can you, @patyork, help with this issue, too?

Please make sure that the boxes below are checked before you submit your issue. If your issue is an implementation question, please ask your question on [StackOverflow](http://stackoverflow.com/questions/tagged/keras) or [join the Keras Slack channel](https://keras-slack-autojoin.herokuapp.com/) and ask there instead of filing a GitHub issue.

Thank you!

- [o ] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps

- [ o] If running on TensorFlow, check that you are up-to-date with the latest version. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [ o] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps

- [o ] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

**Comment by gvtulder** [comment]
This seems to be fixed in the Theano master branch.
